From ef647cce7328b12c3f734f3ebd373bf577de27ca Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 29 Oct 2015 15:30:53 +0000
Subject: [PATCH] abuild: do not install itself as a makedepend

we might need add $pkgname as a dependency for $pkgname-dev and then we
should not install ourselves in case makedepends="$depends_dev"

This is needed foor bootstraping.
---
 abuild.in | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/abuild.in b/abuild.in
index 0dd8f9a..e44696b 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1721,14 +1721,12 @@ builddeps() {
 			[ "$pkgname" = "${i%%[<>=]*}" ] && continue
 			subpackages_has ${i%%[<>=]*} || builddeps="$builddeps $i"
 		done
-		hostdeps="$makedepends_host"
-		for i in $depends; do
+		for i in $depends $makedepends_host; do
 			[ "$pkgname" = "${i%%[<>=]*}" ] && continue
 			subpackages_has ${i%%[<>=]*} || hostdeps="$hostdeps $i"
 		done
 	else
-		builddeps="$makedepends"
-		for i in $BUILD_BASE $depends; do
+		for i in $BUILD_BASE $depends $makedepends; do
 			[ "$pkgname" = "${i%%[<>=]*}" ] && continue
 			subpackages_has ${i%%[<>=]*} || builddeps="$builddeps $i"
 		done
@@ -1962,9 +1960,8 @@ post_add() {
 }
 
 deps() {
-	local builddeps i
-	builddeps="$makedepends"
-	for i in $depends; do
+	local builddeps= i
+	for i in $depends $makedepends; do
 		[ "$pkgname" = "${i%%[<>=]*}" ] && continue
 		subpackages_has ${i%%[<>=]*} || builddeps="$builddeps $i"
 	done
-- 
2.6.1

